keras/layers/core.py

```diff
@@ -589,6 +589,27 @@ def get_config(self):
                 "output_reconstruction": self.output_reconstruction}
 
 
+class Maxout(Layer):
+    '''
+        Max-out layer, nb_feature is the number of pieces in the piecewise linear approx.
+        nb_feature must divide the first dimension of the input.
+    '''
+    def __init__(self, nb_feature=4):
+        super(Maxout, self).__init__()
+        self.nb_feature = int(nb_feature)
+
+    def get_output(self, train=False):
+        X = self.get_input(train)
+        nshape = T.concatenate([(X.shape[0], self.nb_feature, X.shape[1] // self.nb_feature),
```

Comment:
>
> A shape being not a tensor but merely a tuple, why instantiate it via T.concatenate instead of simple tuple addition?
>
> I could not get it to work using tuple addition. Will be glad to learn an
alternative.
Here is what happened when I tried tuple addition.

numpy tuple addition:

In [19]: x = numpy.ones((5,4,3,2))


In [20]: w = (x.shape[0], 2, x.shape[1]//2) + x.shape[2:]


In [21]: w

Out[21]: (5, 2, 2, 3, 2)


Theano tuple addition:


In [22]: x = T.tensor4('x', 'float32')


In [23]: w = (x.shape[0], 2, x.shape[1]//2) + x.shape[2:]


In [24]: f = theano.function([x], w)


In [25]: f(numpy.ones((5,4,3,2), dtype='float32'))


ValueError: Input dimension mis-match. (input[0].shape[0] = 3,
input[1].shape[0] = 2)

Apply node that caused the error: Elemwise{Add}[(0, 0)](MakeVector.0,
MakeVector.0)

Inputs types: [TensorType(int64, vector), TensorType(int64, vector)]

Inputs shapes: [(3,), (2,)]

Inputs strides: [(8,), (8,)]

Inputs values: [array([5, 2, 2]), array([3, 2])]